### ISSUE TITLE
[Feature] Add AdamW-CPU operator to support offload calculation

### ISSUE THREAD

**Issue by Baibaifan** [PaddlePaddle/Paddle#41950]
### 需求描述 Feature Description

When training a large model and using the AdamW optimizer, the optimizer parameters/status are often offloaded to the cpu memory, and then used to update the parameters on the cpu. At present, the AdamW-CPU implemented by Paddle has poor performance, and I hope to provide higher performance. The AdamW-CPU implementation improves the overall computing speed.

We hope to give the high-performance version of AdamW-CPU based on the optimized Adam of the previous version. 
And we hope to give the performance comparison data of the AdamW-CPU, and compare it with the performance comparison data in part 6.2.4 of the following paper.

We hope to supplement the high-performance comparison data of Adam-CPU, AdamW-CPU with Adam-CPU, AdamW-CPU in Pytorch.
Such as:
<img width="433" alt="image" src="https://user-images.githubusercontent.com/39549453/163818385-8073e1eb-54cb-4a6f-b431-2cbcc020f81a.png">

reference: Section 5.1 in the paper
https://arxiv.org/pdf/2101.06840.pdf

**Comment by Aganlengzi** [comment]
@lidanqing-intel This feature is like https://github.com/PaddlePaddle/Paddle/issues/39005 and need comparison data.

**Comment by wozna** [comment]
Hi @Baibaifan. We are happy that you used the Adam-CPU implementation. 
For now, we added a high-performance version of Adam-CPU (https://github.com/PaddlePaddle/Paddle/pull/39158), but this version will **be used only if the machine supports AVX512 instructions** (AVX512F flag is on)
In AdamW, computations that are the same as in Adam, used this high-performance version, but only this different part where params are adjusted is not optimized yet https://github.com/PaddlePaddle/Paddle/blob/34ac7b74c216bd02d44d9bc57b1537343adc0934/paddle/phi/kernels/funcs/adam_functors.h#L426 
Could you tell me what machine you used for those Adam and AdamW tests? 
And can you share the information on which model it is checked or test we can use to do benchmarking?

**Comment by Baibaifan** [comment]
We confirm that our machine supports AVX512 (AVX512F flag is on)，We want to test the performance comparison of Adam with a fixed size parameter between Pytorch's Adam and high-performance version Adam.

**Comment by Aganlengzi** [comment]
> We confirm that our machine supports AVX512 (AVX512F flag is on)，We want to test the performance comparison of Adam with a fixed size parameter between Pytorch's Adam and high-performance version Adam.

@wozna hi, Please refer to @Baibaifan's comment, operator level performance comparison is enough.

**Comment by Baibaifan** [comment]
@wozna  Hi, I think V100-32G can be used for testing, and the performance comparison of Adam and AdamW for 5000 times parameters optimization with fixed size parameters and gradients between Pytorch and high-performance version.

**Comment by wozna** [comment]
@Baibaifan @Aganlengzi I prepared AdamW-CPU JIT kernel https://github.com/PaddlePaddle/Paddle/pull/42522. 
I tested only 1 epoch for vgg model with results  

Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz

Average kernel time [ms]
| Threads | AdamW with Eigen | AdamW JIT |
|---------|------------------|-----------|
| 1       | 0.0667           | 0.0655724 |
| 20      | 0.133347         | 0.118731  |

Please test it and let me know how the kernel works in your case.

**Comment by lidanqing-vv** [comment]
#42522 has been merged, hence I close this issue. @wozna Good job Joanna!
